@dotnet/aspnet-build @dotnet/dnceng I posted about this on our internal "ASP.NET Build" teams chat, but I'll ask here too for wider visibility.

We are having an issue updating our project templates to reference the latest version of [Microsoft.Identity.Web 2.15.2](https://www.nuget.org/packages/Microsoft.Identity.Web/2.15.2#dependencies-body-tab) because it depends on [Microsoft.Identity.Web.TokenAcquisition 2.15.2](https://www.nuget.org/packages/Microsoft.Identity.Web.TokenAcquisition/2.15.2#dependencies-body-tab) which in turn depends on RC2 packages that we ship like Microsoft.AspNetCore.Authentication.OpenIdConnect (>= 8.0.0-rc.2.23480.2).
 
The problem is that our template tests see our Microsoft.AspNetCore.Authentication.OpenIdConnect 8.0.0-ci artifacts as a forced downgrade and fails our template tests with a build error like:
 
```
/datadisks/disk1/work/BCE909F9/w/B40D0985/e/Templates/BaseFolder/AspNet.vxrmciskazme/AspNet.vxrmciskazme.csproj : error NU1605: Warning As Error: Detected package downgrade: Microsoft.AspNetCore.Authentication.JwtBearer from 8.0.0-rc.2.23480.2 to 8.0.0-ci. Reference the package directly from the project to select a different version.
/datadisks/disk1/work/BCE909F9/w/B40D0985/e/Templates/BaseFolder/AspNet.vxrmciskazme/AspNet.vxrmciskazme.csproj : error NU1605:  AspNet.vxrmciskazme -> Microsoft.Identity.Web 2.15.2 -> Microsoft.Identity.Web.TokenAcquisition 2.15.2 -> Microsoft.AspNetCore.Authentication.JwtBearer (>= 8.0.0-rc.2.23480.2)
/datadisks/disk1/work/BCE909F9/w/B40D0985/e/Templates/BaseFolder/AspNet.vxrmciskazme/AspNet.vxrmciskazme.csproj : error NU1605:  AspNet.vxrmciskazme -> Microsoft.AspNetCore.Authentication.JwtBearer (>= 8.0.0-ci)
```

https://dev.azure.com/dnceng-public/public/_build/results?buildId=442668&view=ms.vss-test-web.build-test-results-tab 

Have we ever referenced an external package in our project templates that reference a prerelease package we built? I assume not, or else we would have run into this issue before. What should we do to fix this?

--- a/src/ProjectTemplates/Web.ProjectTemplates/WebApi-CSharp.csproj.in
+++ b/src/ProjectTemplates/Web.ProjectTemplates/WebApi-CSharp.csproj.in
@@ -12,9 +12,9 @@
 
   <!--#if (OrganizationalAuth || IndividualB2CAuth || EnableOpenAPI || WindowsAuth) -->
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="${MicrosoftAspNetCoreAuthenticationJwtBearerVersion}" Condition="'$(OrganizationalAuth)' == 'True' OR '$(IndividualB2CAuth)' == 'True'" />
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="${MicrosoftAspNetCoreAuthenticationJwtBearerVersion}" Condition="'$(OrganizationalAuth)' == 'True' OR '$(IndividualB2CAuth)' == 'True'" NoWarn="NU1605" />
     <PackageReference Include="Microsoft.AspNetCore.Authentication.Negotiate" Version="${MicrosoftAspNetCoreAuthenticationNegotiateVersion}" Condition="'$(WindowsAuth)' == 'True'" />
-    <PackageReference Include="Microsoft.AspNetCore.Authentication.OpenIdConnect" Version="${MicrosoftAspNetCoreAuthenticationOpenIdConnectVersion}" Condition="'$(OrganizationalAuth)' == 'True' OR '$(IndividualB2CAuth)' == 'True'" />
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.OpenIdConnect" Version="${MicrosoftAspNetCoreAuthenticationOpenIdConnectVersion}" Condition="'$(OrganizationalAuth)' == 'True' OR '$(IndividualB2CAuth)' == 'True'" NoWarn="NU1605" />
     <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="${MicrosoftAspNetCoreOpenApiVersion}" Condition="'$(EnableOpenAPI)' == 'True' AND '$(UsingMinimalAPIs)' == 'True'" />
     <PackageReference Include="Microsoft.Identity.Web" Version="${MicrosoftIdentityWebVersion}" Condition="'$(OrganizationalAuth)' == 'True' OR '$(IndividualB2CAuth)' == 'True'"/>
     <PackageReference Include="Microsoft.Identity.Web.GraphServiceClient" Version="${MicrosoftIdentityWebGraphServiceClientVersion}" Condition=" '$(GenerateGraph)' == 'True' " />

